applications/dashboard/controllers/Api/UsersApiController.php

```diff
@@ -285,6 +291,72 @@ public function post(array $body) {
     }
 
     /**
+     * Submit a new user registration.
+     *
+     * @param array $body The request body.
+     * @throws ClientException if terms of service field is false.
+     * @throws ServerException if an unknown error was encountered when creating the user.
+     * @throws ValidationException if the registration is flagged as SPAM, but no discoveryText was provided.
+     * @return Data
+     */
+    public function post_register(array $body) {
+        $this->permission();
+
+        $registrationMethod = $this->configuration->get('Garden.Registration.Method');
+        $registrationMethod = strtolower($registrationMethod);
+
+        $userData = $this->caseScheme->convertArrayKeys($body);
+
+        $inputProperties = [
+            'email:s' => 'An email address for this user.',
+            'name:s' => 'The username.',
+            'password:s' => 'A password for this user.',
+            'termsOfService:b' => 'Were the terms of use accepted?'
+        ];
+        if ($registrationMethod === 'invitation') {
+            $inputProperties['invitationCode:s'] = 'An invitation code for registering on the site.';
+        } elseif ($this->userModel->isRegistrationSpam($userData)) {
```

Comment:
We can't change the API spec depending on what is sent to it because we will never be able to generate the documentation properly if we do this.

We should put `discoveryText:s?` in the inputProperties and incidate that this field will be required if the response is a 202. THEN we can remove the ? if the $userData is flagged as spam which will make the validation work as intended.